src/Context/ThemeContext.tsx

```diff
@@ -1,27 +1,39 @@
 
-import { createContext, useState , ReactNode} from 'react';
+import { createContext, useState , ReactNode, useEffect} from 'react';
 
 export interface ITheme{
-    theme: string;
-    setTheme: React.Dispatch<React.SetStateAction<string>>;
+    isDarkMode: boolean;
+    setIsDarkMode: React.Dispatch<React.SetStateAction<boolean>>;
     toggleTheme: () => void;
+    themeClass: string;
 }
 interface ThemeContextProviderProps {
     children: ReactNode;
   }
 export const ThemeContext = createContext<ITheme | null>(null);
 
 const ThemeContextProvider: React.FC<ThemeContextProviderProps>= (props)=>{
-    const [theme, setTheme] = useState(()=> localStorage.getItem("theme") || "light");
-    // const [isDarkMode, setIsDarkMode] = useState(false); 
+    const [isDarkMode, setIsDarkMode] = useState(() => {
+        const savedTheme =localStorage.getItem("theme");
+        return savedTheme === 'dark';
+    });
    const toggleTheme = () =>{
-        setTheme((current)=> (current === 'light' ? 'dark' : 'light' ))
+        // setTheme((current)=> (current === 'light' ? 'dark' : 'light' ))
+        setIsDarkMode((previousValue) => !previousValue); 
     }
+    useEffect(() => {
+        // Save the theme preference to localStorage
+        localStorage.setItem("theme", isDarkMode ? "dark" : "light");
```

Comment:
You can store the item in the local storage directly as isDarkMode and save its boolean value instead of this conversion between names and boolean values
`localStorage.setItem("isDarkMode", isDarkMode )`

and use it directly in your application.

Reply:
and this will enable you to simplify the initialization function in the useState as follow:
` const [isDarkMode, setIsDarkMode] = useState(() =>  localStorage.getItem("isDarkMode")`

Reply:
done

Reply:
> done

I tested the application now but there is a small bug. 
Tf we switch to dark mode and then refresh the page, it returns back to light-mode not dark-mode